### PR TITLE
fix: update SCORER usage sklearn version > 1.2.2

### PR DESCRIPTION
**Pull request by serena-ruan**
## Related Issues/PRs

 #xxx

## What changes are proposed in this pull request?
sklearn > 1.2.2 removes sklearn.metrics.SCORERS, thus changing the usage in mlfllow.sklearn.autolog

## How is this patch tested?

- [x] Existing unit/integration tests
- [ ] New unit/integration tests
- [ ] Manual tests (describe details, including test results, below)

## Does this PR change the documentation?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Make sure the changed pages / sections render correctly in the documentation preview.

### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [X] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: mlflow/sklearn/__init__.py [hunk 1]
```diff
@@ -1787,8 +1787,13 @@ def out(*args, **kwargs):
...
 
-        for scorer in sklearn.metrics.SCORERS.values():
-            safe_patch(flavor_name, scorer, "_score_func", patched_metric_api, manage_run=False)
+        if Version(sklearn.__version__) > Version("1.2.2"):
+            for scoring in sklearn.metrics.get_scorer_names():
+                scorer = sklearn.metrics.get_scorer(scoring)
+                safe_patch(flavor_name, scorer, "_score_func", patched_metric_api, manage_run=False)
+        else:
+            for scorer in sklearn.metrics.SCORERS.values():
+                safe_patch(flavor_name, scorer, "_score_func", patched_metric_api, manage_run=False)
```
Comment (WeichenXu123) [inline mlflow/sklearn/__init__.py:1796]: you don't need to duplicate the for-loop body code,

you can write:
```
if (Version(sklearn.__version__) > Version("1.2.2")
  sklearn_scorer_names = ...
else:
  sklearn_scorer_names = ...

for scorer in sklearn_scorer_names:
  ...
```

**Review by WeichenXu123 — APPROVED** [review]

File: mlflow/sklearn/__init__.py (hunk 1 above)
Comment (serena-ruan) [reply mlflow/sklearn/__init__.py:1796]: We're iterating over the scorers not the names. If iterating over the names, we need to either use 'sklearn.metrics.get_scorer(name)' to fetch it in version >1.2.2 or 'SCORERS[name]' for versions <=1.2.2

Comment (WeichenXu123) [reply mlflow/sklearn/__init__.py:1796]: oh I miss it.

**Review by BenWilson2 — APPROVED** [review]
LGTM! Thanks @serena-ruan :)

**Pull request merged**